tests/gui/test_widgets.py

```diff
@@ -950,6 +950,70 @@ def test_MainView_set_conversation(mocker):
     mv.view_layout.addWidget.assert_called_once_with(mock_widget)
 
 
+def test_MainView__delete_source(mocker):
+    """
+    Ensure that the source and conversation provided by source uuid are deleted.
+    """
+    source = factory.Source(uuid="123")
+    conversation_wrapper = SourceConversationWrapper(source, mocker.MagicMock())
+    mv = MainView(None)
+    mv.source_conversations = {}
+    mv.source_conversations["123"] = conversation_wrapper
+    mv.source_list.controller = mocker.MagicMock()
+    mv.source_list.update([source])
+    assert not mv.source_list.source_items == {}
+
+    mv._delete_source("123")
+
+    assert mv.source_conversations == {}
+    assert mv.source_list.source_items == {}
```

Comment:
Python question / maybe nitpicking: is the intention to check that the collections are not empty, then empty? And could a `len` check work/convey that better?

I'm thinking something in the lines of:

```diff
- assert not mv.source_list.source_items == {}
+ assert len(mv.source_list.source_items) > 0

- assert mv.source_list.source_items == {}
+ assert len(mv.source_list.source_items) == 0

Reply:
i agree, that's cleaner